src/raylib.cr

```diff
@@ -81,6 +81,23 @@ module RaylibAdditions
       end
       return Raylib.gen_image_color(64, 64, Raylib::PURPLE)
     end
+
+    # Exports a texture to a png given a *texture_name*, a *filename*, and a *palette*
+    #
+    # ```
+    # my_wad = WAD.read("Path/To/Wad")
+    #
+    # palette = my_wad.playpal.palettes[0]
+    #
+    # my_wad.export_texture("MyTexture", "Path/To/MyGraphic.png", palette)
+    # ```
+    def export_texture(texture_name : String, filename : String | Path, palette : ::WAD::Playpal::Palette)
```

Comment:
I see you using `::`to access the top level `::WAD::Playpal::Palette` but you don't need to. When looking for a class name, the class checking order goes something like self, subclasses, superclasses, toplevel. This happens any time you check for methods, modules, classes, enums, etc, etc, that are inside another class.

Reply:
Also good job on the `filename : String | Path`